vertical-pod-autoscaler/recommender/recommender.go

```diff
@@ -35,13 +39,59 @@ type Recommender interface {
 }
 
 type recommender struct {
+	clusterState            model.ClusterState
 	specClient              cluster.SpecClient
 	metricsFetchingInterval time.Duration
+	prometheusClient        signals.PrometheusClient
```

Comment:
can we have an interface that will hide prometheus details from recommender?

Reply:
Left a TODO for now, there's plenty of work and I do not want to have a single PR that makes world perfect.